xbmc/addons/Repository.cpp

```diff
@@ -250,14 +246,23 @@ bool CRepositoryUpdateJob::DoWork()
 
     for (const auto& addon : addons)
     {
-      if (!addon->Props().fanart.empty())
-        textureDB.InvalidateCachedTexture(addon->Props().fanart);
-      if (!addon->Props().icon.empty())
-        textureDB.InvalidateCachedTexture(addon->Props().icon);
+      AddonPtr oldAddon;
+      if (database.GetAddon(addon->ID(), oldAddon) && addon->Version() > oldAddon->Version())
+      {
+        if (!addon->Props().icon.empty())
+        {
+          CLog::Log(LOGDEBUG, "CRepository: invalidating cached art for '%s'", addon->ID().c_str());
```

Comment:
Don't we want this log line if either the icon or the fanart is re-cached? Otherwise it might be misleading. Or do you assume that every addon has at least an icon but not necessarily a fanart?